### PR TITLE
Ftr: registry ip:port set from enviroment variable

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
**What this PR does**:
Add support of "setting registry port and ip that read from environment". 
- In the past, registry ip can only be set to server local ip, whch can't set to specific ip:port by developer under the needs of specific scenarios. 
- This feature is supported in dubbo-java
I use this feature in design of [docker example](https://github.com/LaurenceLiZhixin/dubbo-go-samples/blob/385de26e859174fa564ab57c92f837b5164a05e4/docker/docker_run.sh#L5)
In the docker environment example, registry ip and port should set by developer's defination, which is also dubbo-java does.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: README_CN.md [hunk 1]
```diff
@@ -180,9 +180,9 @@ make test
 <a href="http://alexstocks.github.io/html/dubbogo.html">
 <img src="./doc/pic/misc/dubbogo-dingding.png"  height="80" width="80"></a>
 
-如果想加入到社区微信群，可以先添加社区负责人 于雨 的微信 AlexanderStocks 。添加微信之前，请先给 dubbo-go 点 star 作为对项目的支持，添加好友时请报上 github ID 以进行验证。
+如果想加入到社区微信群，可以先添加社区负责人 于雨 的微信 AlexanderStocks 。
```
Comment (zouyx) [inline README_CN.md:183]: Why change this two lines?

**Review by Patrick0308 — COMMENTED** [review]
LGTM

File: README_CN.md (hunk 1 above)
Comment (AlexStocks) [reply README_CN.md:183]: refer from dev.

**Comment by AlexStocks** [comment]
> LGTM

approve instead.

File: registry/base_registry.go [hunk 2]
```diff
@@ -136,6 +136,13 @@ func (r *BaseRegistry) Register(conf *common.URL) error {
 		ok  bool
 		err error
 	)
+	// if developer define registry port and ip, use it first.
+	if ipToRegistry := os.Getenv("DUBBO_IP_TO_REGISTRY"); ipToRegistry != "" {
```
Comment (gaoxinge) [inline registry/base_registry.go:140]: Why are ip and port from environment instead of configuration?

Comment (LaurenceLiZhixin) [reply registry/base_registry.go:140]: Because this feature is supported in dubbo-java, and I use this feature in design of docker example

Comment (gaoxinge) [reply registry/base_registry.go:140]: ok

**Review by gaoxinge — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**